File: 프로그래머스/lv0/120844. 배열 회전시키기/배열 회전시키기.cs
using System;

public class Solution {
    public int[] solution(int[] numbers, string direction) {
        
        int size = numbers.Length;
        int[] answer = new int[size];
        
        int offset = direction == "left" ? 1 : -1;
        
        for(int i = 0; i < size; i++)
        {
            int offsetIndex = (i + offset + size) % size;
            answer[i] = numbers[offsetIndex];
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> numbers, string direction) {
    
    vector<int> answer;
    
    if (direction == "left")
    {
        answer = vector<int>(numbers.begin() + 1, numbers.end());
        answer.push_back(numbers.front());
    }
    else
    {
        answer.push_back(numbers.back());
        answer.insert(answer.end(), numbers.begin(), numbers.end() - 1);
    }
    
    return answer;
}